### PR TITLE
Support ES2015

### PR DESCRIPTION
**Pull request by LesikDee**
Classes were created, added usage let/const instead of var, some useless functions were removed  

## Motivation and Context

Code base supports old standard of JS, standard was changed to ES2015

## How Has This Been Tested?

e2e tests passed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/io/parsers/CIFParser.js [hunk 1]
```diff
@@ -60,753 +59,755 @@ function nameToElement(name) {
...
-  this.molecules = [];
-  this._options.fileType = 'cif';
+class AtomDataError extends Error {
+  constructor(message) {
+    super();
+    this.name = 'AtomDataError';
+    this.message = message;
+  }
 }
+//AtomDataError.prototype = Object.create(Error.prototype);
```
Comment (paulsmirnov) [inline src/io/parsers/CIFParser.js]: Should we remove the commented line?

Note that such inheritance didn't work correclty in Babel 6 (see https://stackoverflow.com/questions/33870684/why-doesnt-instanceof-work-on-instances-of-error-subclasses-under-babel-node/33877501#33877501). I've updated to Babel 7 yesterday so it should now work correctly (https://babeljs.io/blog/2018/08/27/7.0.0#class-c-extends-htmlelement).

File: src/io/parsers/CIFParser.js [hunk 2]
```diff
@@ -60,753 +59,755 @@ function nameToElement(name) {
...
+      for (let r = start[1]; r <= end[1]; ++r) {
+        residues[r]._secondary = struct;
+      }
     }
-    ops[idc[i]] = mtx;
   }
 
-  return ops;
-}
+  static _getOperations(operList) {
```
Comment (paulsmirnov) [inline src/io/parsers/CIFParser.js]: Do we need this as a class member or should leave it be as a free function? Please check other such functions in this file and in others.

File: src/io/parsers/CCP4Parser.js [hunk 3]
```diff
@@ -2,213 +2,200 @@ import _ from 'lodash';
 import Parser from './Parser';
 import chem from '../../chem';
 import * as THREE from 'three';
-import utils from '../../utils';
+//import utils from '../../utils';
```
Comment (paulsmirnov) [inline src/io/parsers/CCP4Parser.js]: Let's remove dead code.

File: src/gfx/VolumeMesh.js [hunk 4]
```diff
@@ -80,35 +78,38 @@ VolumeMesh.prototype._updateVertices = (function() {
     [3, 7, 1, 1, 0]
   ];
 
-  var edgeIntersections = [];
-  for (var j = 0; j < 12; ++j) {
-    edgeIntersections.push(new THREE.Vector3());
-  }
-
+  static _edgeIntersections = [];
```
Comment (paulsmirnov) [inline src/gfx/VolumeMesh.js]: In the old code version this array was initialized only once in order to improve performance.

File: src/gfx/VolumeMesh.js [hunk 5]
```diff
@@ -80,35 +78,38 @@ VolumeMesh.prototype._updateVertices = (function() {
...
-    var cornerMark = [0, 0, 0, 0, 0, 0, 0, 0];
-    var edgeMark = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
+    const vert = this.vertices;
+    const size = this.size;
 
-    const curEdge = new THREE.Vector3();
+    const cornerMark = [0, 0, 0, 0, 0, 0, 0, 0];
+    const edgeMark = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
+
+    let curEdge = new THREE.Vector3();
```
Comment (paulsmirnov) [inline src/gfx/VolumeMesh.js]: It was a `const` before.

File: src/gfx/Stats.js [hunk 6]
```diff
@@ -2,39 +2,35 @@
...
+    this._deltas = new Array(20);
+    this._index = 0;
+    this._total = 0.0;
+    this._count = 0;
+  }
 
-  begin: function() {
-    this._startTime = now();
-  },
+  //begin() { this._startTime = now(); }
```
Comment (paulsmirnov) [inline src/gfx/Stats.js]: Is there a reason to remove this method?

File: src/io/parsers/CIFParser.js [hunk 7]
```diff
@@ -60,753 +59,755 @@ function nameToElement(name) {
...
+    return typeof data === 'string' && Parser.checkDataTypeOptions(options, 'cif');
   }
 
-  // reorganize molecules for complex and check chains
-  var molecules = complex.getMolecules();
-  for (i = 0; i < count; i++) {
-    var molecule = this.molecules[i];
-    molecules[i] = new Molecule(complex, molecule.name, i + 1);
-    molecules[i]._residues = molecule.residues;
+  canProbablyParse(data) {
```
[the remark this answers is not in the document]
Comment (LesikDee) [reply src/io/parsers/CIFParser.js]: It seems i somehow missed this constructions  in a few files. 
Checked all files and fixed the necessarry ones.

File: src/io/parsers/MMTFParser.js [hunk 8]
```diff
@@ -20,458 +20,464 @@ var
...
+    if (!data) {
+      return false;
+    }
+    return data instanceof ArrayBuffer && Parser.checkDataTypeOptions(options, 'mmtf');
+  }
 
-MMTFParser.prototype._onChain = function(chainData) {
-  if (chainData.modelIndex !== 0) {
-    return;
+  static getFirstByte(buf) {
```
[the remark this answers is not in the document]
Comment (LesikDee) [reply src/io/parsers/MMTFParser.js]: I think semantically it`d be better if we leave this as a free functions.
Did the same in the other files.

File: src/gfx/Stats.js (hunk 6 above)
Comment (LesikDee) [reply src/gfx/Stats.js]: Yes, this method is unused and its realization is in the constructor.

File: src/io/parsers/CIFParser.js (hunk 1 above)
Comment (LesikDee) [reply src/io/parsers/CIFParser.js]: After your update i remove this line with a clear conscience.

**Comment by LesikDee** [comment]
Thank you for your corrections. These mistakes are fixed.

**Pull request merged**